u_common.go

```diff
@@ -216,6 +220,50 @@ var (
 	HelloQQ_11_1 = ClientHelloID{helloQQ, "11.1", nil}
 )
 
+// The key is Version(string) and the value is Weights(struct).
+// Store your weights into this map and uTLS will load them.
+var WeightsMap sync.Map
+
+// Weights are used in func generateRandomizedSpec().
+type Weights struct {
+	Weight_Extensions_Append_ALPN                             float64
+	Weight_TLSVersMax_Set_VersionTLS13                        float64
+	Weight_CipherSuites_Remove_RandomCiphers                  float64
+	Weight_SigAndHashAlgos_Append_ECDSAWithSHA1               float64
+	Weight_SigAndHashAlgos_Append_ECDSAWithP521AndSHA512      float64
+	Weight_SigAndHashAlgos_Append_PSSWithSHA256               float64
+	Weight_SigAndHashAlgos_Append_PSSWithSHA384_PSSWithSHA512 float64
+	Weight_CurveIDs_Append_X25519                             float64
+	Weight_CurveIDs_Append_CurveP521                          float64
+	Weight_Extensions_Append_Padding                          float64
+	Weight_Extensions_Append_Status                           float64
+	Weight_Extensions_Append_SCT                              float64
+	Weight_Extensions_Append_Reneg                            float64
+	Weight_Extensions_Append_EMS                              float64
+	Weight_OnlyKeyShare_Set_CurveP256                         float64
+	Weight_Extensions_Append_ALPS                             float64
+}
+
+// Please make a copy first instead of modifying them directly.
+var DefaultWeights = Weights{
+	Weight_Extensions_Append_ALPN:                             0.7,
+	Weight_TLSVersMax_Set_VersionTLS13:                        0.4,
```

Comment:
As discussed if we could update this list of weights it would be great. For example, perhaps it is a good time to have TLS 1.3 to weigh more?

Reply:
Maybe we should update these preset weights in another PR?

Reply:
Sure. I was about to contact Sergey to learn how did he yield all these parameters.